### DISCUSSION TITLE
Does hidden props works for .findOne() or .findOneOrFail() repositories functions ?

### DISCUSSION THREAD

**Discussion by Luxiorawa** [mikro-orm/mikro-orm#3556] [Q&A] [answered]
Hey, i have a question about something i just found. 

I have a User entity that can be simplified like this : 

```ts
export class User {
  @PrimaryKey()
  public id: number;

 @Property({ length: 50 })
  public login: string;

  @Property({
    hidden: true
  })
  public password: string;
}
```

I also use repositories to access any datas in my services. When i'm using find() function, the password field is not there. 

I'm calling this : 

```ts
    return await this.usersRepository.find({});
```

But when i'm calling this : 

```ts
    return await this.usersRepository.findOneOrFail({
        id: userId,
    });
```

It is returning password field. I need to add lazy: true to my entity so that it doesn't appear at all. Is that a wanted process ? seems pretty weird to me, but maybe there is things i don't understand.

**Comment by B4nan** [comment, marked as answer]
Both calls will return it, the flag is used only for serialization, you need to look at that instead your entity instance. As opposed to this, lazy props mean it wont be even selected.

**Reply by Luxiorawa** [reply]
Oh i see, make sense now, thanks for the answer !
